app/models/job.ts

```diff
@@ -443,7 +507,10 @@ export class Job extends Record {
     this.validateStatus();
     this.message = truncateString(this.message, 4096);
     this.request = truncateString(this.request, 4096);
-    const { links, originalStatus } = this;
+    const { links, originalStatus, collectionIds } = this;
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
```

Comment:
This is not pretty but it's the best solution I could find, without resorting to using a bunch or '|' in the Job classes/interfaces.

Reply:
Using '|' might be better?

Reply:
I went with a different strategy for JobLink (feel free to ignore this):

I added an interface `JobLinkOrRecord` that allowed for multiple types and then I performed the transformations in the `save` call and the constructor. That way I wasn't mucking around with setting things on `this` and then setting them back.

I don't entirely like that approach either.